node_api/src/logic/validator.ts

```diff
@@ -85,6 +86,50 @@ export default class Validator {
     return { isValid: valid, message: message };
   }
 
+  private isValidProperties(data: any) {
+    let valid = true;
+    let message: string | undefined = undefined;
+    const props: any[] = data[SHACL.PROPERTY];
+
+    if (!props || props.length == 0) {
+      valid = false;
+      message = "Data models must have at least 1 property";
+    } else {
+      for (const prop of props) {
+        if (!prop[SHACL.PATH]) {
+          valid = false;
+        } else {
+          const path = prop[SHACL.PATH].length ? prop[SHACL.PATH][0] : prop[SHACL.PATH];
+          if (!path?.["@id"]) {
+            valid = false;
+          }
+        }
+
+        if (prop[SHACL.NODE]) {
+          const range = prop[SHACL.NODE].length ? prop[SHACL.NODE][0] : prop[SHACL.NODE];
```

Comment:
this pattern is repeated in all predicates of property

const range = prop[SHACL.NODE].length ? prop[SHACL.NODE][0] : prop[SHACL.NODE];
          if (!range?.["@id"]) {
            valid = false;
          }

is it worth extracting it into a separate method with a param of the predicate being validated?
something like isValidPredicate(SHACL.NODE)
or even as a set of parameters isValidPredicate([SHACL.NODE, SHACL.DATATYPE])